rosidl_parser/rosidl_parser/definition.py

```diff
@@ -56,49 +101,64 @@
 
 
 class AbstractType:
-    """The base class for all types."""
+    """The abstract base class for all types."""
 
     __slots__ = ()
 
     def __eq__(self, other):
         return type(self) == type(other)
 
 
-class BaseType(AbstractType):
-    """The base class for types which can be used inside nested types."""
+class AbstractNestableType(AbstractType):
```

Comment:
Any class which is not supposed to instantiated directly starts with `Abstract` now.

Reply:
Instead of using the fuzzy name `BaseType` the term `Nestable` should be clearer. Also the docblock elaborates on nested and nestable types.